File: ParadigmaComp/15Questoes/python/questao11.py
from abc import ABC, abstractmethod
#Classes Abstratas
class Funcionario(ABC):
    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, horas, valor_hora):
        self.horas = horas
        self.valor_hora = valor_hora

    def calcular_salario(self):
        return self.horas * self.valor_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, salario_fixo):
        self.salario_fixo = salario_fixo

    def calcular_salario(self):
        return self.salario_fixo

# Exemplo de uso
horista = FuncionarioHorista(160, 20)
assalariado = FuncionarioAssalariado(3000)

print(horista.calcular_salario())  # Salário do horista
print(assalariado.calcular_salario())  # Salário do assalariado

File: ParadigmaComp/15Questoes/golang/questao11.go
package main
//Classes Abstratas
import "fmt"

// Interface Funcionario
type Funcionario interface {
	CalcularSalario() float64
}

// Estrutura FuncionarioHorista
type FuncionarioHorista struct {
	Horas      int
	ValorHora  float64
}

func (f FuncionarioHorista) CalcularSalario() float64 {
	return float64(f.Horas) * f.ValorHora
}

// Estrutura FuncionarioAssalariado
type FuncionarioAssalariado struct {
	SalarioFixo float64
}

func (f FuncionarioAssalariado) CalcularSalario() float64 {
	return f.SalarioFixo
}

func main() {
	horista := FuncionarioHorista{Horas: 160, ValorHora: 20}
	assalariado := FuncionarioAssalariado{SalarioFixo: 3000}

	fmt.Println(horista.CalcularSalario())  // Salário do horista
	fmt.Println(assalariado.CalcularSalario())  // Salário do assalariado
}
